### PR TITLE
refactored code

### PR DESCRIPTION
**Pull request by vivekmenezes**

### CODE REVIEW

**Comment by bdarnell** [comment]
LGTM

**Comment by petermattis** [comment]
LGTM

File: examples/bank/bank.go [hunk 1]
```diff
@@ -117,9 +117,6 @@ func (bank *Bank) continuousMoneyTransfer(cash int64) {
 			if err := runner.Run(batchRead); err != nil {
 				return err
 			}
-			if batchRead.Results[0].Err != nil {
```
Comment (tamird) [inline examples/bank/bank.go]: can you help me understand why this is OK to remove?

File: examples/bank/bank.go [hunk 2]
```diff
@@ -149,11 +146,13 @@ func (bank *Bank) continuousMoneyTransfer(cash int64) {
 			}
 			return runner.Run(batchWrite)
 		}
+		var err error
 		if *useTransaction {
-			if err := bank.db.Tx(func(tx *client.Tx) error { return transferMoney(tx) }); err != nil {
-				log.Fatal(err)
-			}
-		} else if err := transferMoney(bank.db); err != nil {
+			err = bank.db.Tx(func(tx *client.Tx) error { return transferMoney(tx) })
```
Comment (tamird) [inline examples/bank/bank.go]: `err = bank.db.Tx(transferMoney)`?

File: examples/bank/bank.go (hunk 1 above)
Comment (petermattis) [inline examples/bank/bank.go]: `Run` returns the first error that occurred in the batch.

Comment (tamird) [inline examples/bank/bank.go]: Cool. Can we remove `Result.Err` entirely?

Comment (petermattis) [inline examples/bank/bank.go]: What if you want to know which operation in a batch caused an error? Not sure why you'd want to know that, but that's the use case for `Result.Err`.

File: examples/bank/bank.go (hunk 2 above)
Comment (vivekmenezes) [inline examples/bank/bank.go]: hey sorry I merged this in without seeing your notes.  I think that should work

Comment (petermattis) [inline examples/bank/bank.go]: Mismatched function types. I don't think it will compile. Who is right?

Comment (tamird) [inline examples/bank/bank.go]: `func (db *DB) Tx(retryable func(tx *Tx) error) error {`

Isn't `retryable` of the same type `func(tx *client.Tx) error`?

Comment (petermattis) [inline examples/bank/bank.go]: `transferMoney` has the signature `func(runner client.Runner) error`.

Comment (tamird) [inline examples/bank/bank.go]: Ah, thanks for the enlightenment.

**Pull request merged**